source/fundamentals/crud/read-operations/change-streams.txt

```diff
@@ -288,3 +288,56 @@ output:
 
 For a list of options, see the `FullDocument <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/changestream/FullDocument.html>`__
 API documentation.
+
+Split Large Change Stream Events
+--------------------------------
+
+Starting in MongoDB 7.0, you can use a ``$changeStreamSplitLargeEvent``
+aggregation stage to split events that exceed 16 MB into smaller fragments.
+``$changeStreamSplitLargeEvent`` returns the fragments sequentially using the
+change stream cursor. Each fragment includes a ``SplitEvent`` object containing
+the following fields:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 35 65
+
+   * - Field
+     - Description
+
+   * - ``fragment``
+     - The index of the fragment, starting at 1
+
+   * - ``of``
+     - The total number of fragments in the event
+
+Enable the ``$changeStreamSplitLargeEvent`` aggregation on your change stream as
+shown in the following example:
+
+.. code-block:: java
+
+   ChangeStreamIterable<Document> changeStream = collection.watch(
+         Arrays.asList(Document.parse("{ $changeStreamSplitLargeEvent: {} }")));
+
+.. note::
+
+   You can have only one ``$changeStreamSplitLargeEvent`` stage in your
+   aggregation pipeline, and it must be the last stage in the pipeline.
+
+You can access the ``SplitEvent`` by using your change stream cursor with the
+``getSplitEvent()`` as follows:
+
+.. code-block:: java
+
+   MongoChangeStreamCursor<ChangeStreamDocument<Document>> cursor = changeStream.cursor();
+   
+   // Application code that creates change stream events
+   // ...
```

Comment:
Suggestion:

I think these lines aren't relevant to the example and should be omitted. The other lines demonstrate the objective, and it's not required for the reader to use the same app that watches for change events to also generate them.